launch/launch/logging/__init__.py

```diff
@@ -300,6 +304,14 @@ def get_logger(name=None):
     return logger
 
 
+@expose_substitution('log_dir')
+def _log_dir(data: Iterable[SomeSubstitutionsType]):
+    if len(data) > 0:
+        raise ValueError('log_dir substitution does not expect any arguments')
+    # Parser expects callable
+    return lambda: launch_config.log_dir, {}
```

Comment:
If there's a concrete case where we know this doesn't work, I can add at test.

Wrapping the return value in a TextSubstitution sounds okay to me. I didn't go for a dedicated substitution initially because it seemed Python scripts wouldn't need to use it (they can just use `launch_config.log_dir` directly).

Reply:
Returning TextSubstitution in e3a994a

Reply:
> If there's a concrete case where we know this doesn't work, I can add at test.

Up to now, the returned value of [ExtractSubstitution.transform()](https://github.com/ros2/launch/blob/bfe5ec2101ce1a5e46e5fadd95af546925350621/launch/launch/frontend/parse_substitution.py#L116) was assumed to be a normalized substitution.
I don't have an example where it breaks off the top of my head, maybe it's fine (?).
@hidmic what do you think?

> Wrapping the return value in a TextSubstitution sounds okay to me

Now I remember that there's code assumig that `TextSubstitution` was "literal text" and not an "embedded substitution", e.g.: [here](https://github.com/ros2/launch/blob/bfe5ec2101ce1a5e46e5fadd95af546925350621/launch/launch/frontend/parse_substitution.py#L138).
So I'm not sure that's a good idea.

>  I didn't go for a dedicated substitution initially because it seemed Python scripts wouldn't need to use it (they can just use launch_config.log_dir directly).

Fair, though I think that applies to other substitutions (`ThisLaunchFile`, `ThisLaunchFileDir`, and maybe some others).

Reply:
A string will pass right through the parser. Still, I'd make it a `TextSubstitution` just to keep things sane.